File: trunk/java/UMLGraph/src/org/daisy/urakawa/properties/channel/ChannelAlreadyExistsException.java
package org.daisy.urakawa.properties.channel;

import org.daisy.urakawa.exceptions.CheckedException;

/**
 * This exception should be thrown when trying to add a Channel
 * which name is already used in the list of current channel.
 */
public class ChannelAlreadyExistsException extends CheckedException {

	/**
	 * 
	 */
	private static final long serialVersionUID = 3085958815065326578L;
}

File: branches/sdk-core-structedit/exception/ChannelAlreadyExistsException.cs
using System;

namespace urakawa.exception
{
    /// <summary>
    /// This exception should be thrown when trying to add a Channel
    /// whose localName is already used in the list of current channels.
    /// </summary>
    public class ChannelAlreadyExistsException : CheckedException
    {
        /// <summary>
        /// Constructor setting the message of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        public ChannelAlreadyExistsException(string msg) : base(msg)
        {
        }

        /// <summary>
        /// Constructor setting the message and inner <see cref="Exception"/> of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        /// <param name="inner">The inner exception</param>
        public ChannelAlreadyExistsException(string msg, Exception inner) : base(msg, inner)
        {
        }
    }
}